File: src/python/new_alias.py
#Fri May  3 17:37:22 PDT 2019
from base_command import base_command
import vcs_db as db 


command_info = { 
    'id'    : 'new_alias' , 
    'rules' : [ "?(create|make) ?(new) alias ?(please)" ] , 
    'vars'  : { "input" : { 'default_value' : False , 
			    'query'         : "What is the input?", 
			    'filter'        : None } , 
		"output" : { 'default_value' : False , 
			     'query'         : "What is the output?", 
			     'filter'        : None}}
}
    

class command(base_command) : 
    
    def __init__(self,opts) : 
        super().__init__(opts) 
        
    def run(self) : 
        self.log.i("Creating new alias") 
        
        #when the program runs we should just validate 
        _in = self.args.input
        _ou = self.args.output

        self.emit( "Confirm that {} will be aliased to {}".format(_in,_ou) ) 
        
        response = self.get_input()
        # validate user result 
        if (response  == "yes") : 
            # the args are correct, so we will send the info to the database 
            result = db.new_alias({'input' : _in , 'output' : _ou })
            self.emit(result)
        elif (response == "no") : 
            self.emit("Exiting")
        else : 
            self.emit("Please say either yes or no. Exiting") 
            
        # end the program 
        self.finish("DONE") 
            
                
       

File: src/nodejs/commands2/builtins/alias_IGNORE_/new_alias/new_alias.js
//Fri May  3 17:37:22 PDT 2019
let vcs = require("../vcs.js")
let id = "new_alias"

class append2file extends vcs.base_command { 
    
    constructor(config) { 
	super({id})
    }
    
    //all commands have a static method returning their info (used for initialization)
    static get_info() { 
	return {
	    id    : id , 
	    rules : [ "?(create|make) ?(new) alias ?(please)", 
		    ],
	    vars     : { "input" : { default_value : false , 
				     query         : "What is the input?", 
				     filter        : null } , 
			 "output" : { default_value : false , 
				     query         : "What is the output?", 
				      filter        : null }} 			 
	}
    } 
    
    //all commands must implement the async run method
    async run() { 
	//loop read from the input channel 
	while(true ) {
	    
	}
	//input channel has been closed 
	this.finish({payload : {result : "OK"}}) 
    } 
    
    //convenice function for appending (used above) 
    append(text) { 
	let fname = vcs.info.resource_dir + this.args.fname
	this.log.i("Appending the following to file: " + fname) 
	this.log.i(text) 
	fs.appendFileSync(fname, text + "\n") 
    }
    
}


module.exports = append2file 
  
